src/main/java/com/advent/of/code/day7/CrabAdjuster.java

```diff
@@ -0,0 +1,38 @@
+package com.advent.of.code.day7;
+
+import java.util.List;
+import java.util.function.BiFunction;
+
+import static java.util.Comparator.comparingInt;
+
+class CrabAdjuster {
+    int calculateAdjustmentCostForFixedBurnRate(List<Integer> positions) {
+        return calculateCost(positions, (currentPosition, positionCandidate) -> Math.abs(currentPosition - positionCandidate));
+    }
+
+    int calculateAdjustmentCostForIncreasingBurnRate(List<Integer> positions) {
+        return calculateCost(positions, (currentPosition, positionCandidate) -> {
+            int n = Math.abs(currentPosition - positionCandidate);
+            return ((1 + n) * n) / 2;
+        });
+    }
+
+    private int calculateCost(List<Integer> positions, BiFunction<Integer, Integer, Integer> currentCostCalculator) {
```

Comment:
Tu muszę poczytać o interfejsie funkcyjnym BiFunction, bo na pierwszy rzut oka nie jest to dla mnie jasne. Widzę, że w wywołaniu metody calculateCost, przekazujemy metodę jako argument przy użyciu wyrażenia lambda... tak to rozumiem. Jak zacząłem to pisać to chyba zacząłem sam sobie odpowiadać na pytanie, co to jest :D
Fajnie przemyślane: to jest to co sugerowałeś zrobić z tym factorialem u mnie?

Reply:
tak dokładnie! :)